target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621_dlink_dir-xx60-a1.dtsi"
+
+/ {
+	compatible = "dlink,dir-3040-a1", "mediatek,mt7621-soc";
+	model = "D-Link DIR-3040 A1";
+};
+
+&wps {
+	gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+};
+
+&leds {
+	usb2_white {
+		label = "white:usb2";
+		gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		trigger-sources = <&ehci_port2>;
+		linux,default-trigger = "usbport";
+	};
+
+	usb3_white {
+		label = "white:usb3";
+		gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		trigger-sources = <&xhci_ehci_port1>;
+		linux,default-trigger = "usbport";
+	};
+
+	wlan2g {
+		label = "white:wlan2g";
+		gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		linux,default-trigger = "phy0radio";
+	};
+
+	wlan5glb {
+		label = "white:wlan5glb";
+		gpios = <&gpio 9 GPIO_ACTIVE_HIGH>;
+		linux,default-trigger = "phy1radio";
+	};
+
+	wlan5ghb {
+		label = "white:wlan5ghb";
+		gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		linux,default-trigger = "phy2radio";
+	};
+};
+
+
+&wifi0 {
+	mediatek,mtd-eeprom = <&factory 0x0>;
+	nvmem-cells = <&macaddr_factory_e000>;
+	nvmem-cell-names = "mac-address";
+	ieee80211-freq-limit = <2400000 6000000>;
+	mac-address-increment = <1>;
```

Comment:
`mac-address-increment` support has been removed recently. Please rebase it and use the new nvmem `fixed-layout` instead. And this model has 3 wireless phys, so I believe the MAC addresses are not correct.

Reply:
Okay, I've updated to answer some of these concerns. I'm not sure about the "3 wireless phys, so I believe the MAC addresses are not correct" part. The dts file for the dir-3060 has 2 given "<&macaddr_factory_e000 N>" lines for two wifi cards. Yes there are 3 phys, but I believe that two of them are handled by one wifi device. Plus, this setup works for the DIR-3040 and the three phys all have different MAC addresses.

Reply:
> this setup works for the DIR-3040 and the three phys all have different MAC addresses.

Are they the same as the stock firmware? Please add some MAC address explanations in the commit message. This is a good example: f3cdc9f9881796794c06f784a2e4790f5ed75d1f

Reply:
So, I don't recall the original MACs for the stock firmware, but my understanding is that the first wireless interface (WI) is two chips that each serve different bands. WI1 chip1 serves 2G and WI1 chip2 serves 5G. This is also seen in the [deviwiki article for the DIR-3040](https://deviwiki.com/wiki/D-Link_DIR-3060_rev_A1). As such, they have to have the same MAC since it's really the same WI. Here is a table that hopefully answers your question.

| use | address | example |
| --- | --- | --- |
| LAN | label | f4:*:61 |
| WAN | label + 4 | f4:*:65 |
| WI1/2g | label + 1 | f4:*:62 |
| WI1/5g | label + 1 | f4:*:62 |
| WI2/5g | label + 3 | f4:*:64 |

The label MAC address was found in Factory, 0xe000

Again, this was allowed for the DIR-3060 dts file, so I don't see an issue for the DIR-3040 dts file.

If this is enough of an explaination, I can include the table in the commit message.

Reply:
Each SSID must have a different MAC, otherwise you will suffer some strange connection issues. I would suggest returning the stock firmware and recording the correct MACs.

Reply:
@Lucky1openwrt  and @iivailo, do either one of you know how to setup the dts file to have a third MAC for the other interface of the abgn+ac radio?

Reply:
I don't have any devices with a DBDC interface to check
I do remember the driver use to change a some bytes this differs from the dlink OEM address
I currently expect the driver to alter the MAC address for the 2nd interface based off the 1st 
changing 1st and 4th bytes as below 
`
MAC addresses on OEM firmware:

lan      factory 0xe000   f4:*:*:a8:*:65  (label)
wan      factory 0xe006   f4:*:*:a8:*:68
2.4 GHz  [not on flash]   f6:*:*:c8:*:66
5.0 GHz  factory 0x4      f4:*:*:a8:*:66

The increment of the 4th byte for the 2.4g address appears to vary.
Reported cases:

       5g                 2.4g         increment
 f4:XX:XX:a8:XX:66  f6:XX:XX:c8:XX:66  +0x20
 x0:xx:xx:68:xx:xx  x2:xx:xx:48:xx:xx  -0x20
 x4:xx:xx:6a:xx:xx  x6:xx:xx:4a:xx:xx  -0x20`

https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=b8168f4716ad64863db8a6a2aba24bbff4664ffe
https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=6639623e75b84de5cf4a657be7a80383309e290d

Reply:
@Lucky1openwrt Thank you!